src/main/java/core/basesyntax/Table.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+public class Table {
+    private int buy;
+    private int supply;
+    private int result;
+
+    public Table(int buy, int supply) {
+        this.buy = buy;
+        this.supply = supply;
+        setResult(buy, supply);
+    }
```

Comment:
The constructor should not call a setter or any method that is not purely assigning the fields. The calculation of 'result' can be moved to where it's needed, or you can directly assign the result of 'supply - buy' to the 'result' field here if it's necessary for the object's state.